packages/core/src/internal/clients/storage/AbstractStorageClient.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  CreateStore,
+  DeleteStore,
+  ListStores,
+  StorageGet,
+  StoragePut,
+  StorageDelete,
+} from '../../../index';
+import {IStorageDataClient} from './IStorageDataClient';
+import {IStorageClient} from '../../../clients/IStorageClient';
+import {IStorageControlClient} from './IStorageControlClient';
+
+export abstract class AbstractStorageClient implements IStorageClient {
+  protected readonly dataClients: IStorageDataClient[];
+  protected readonly controlClient: IStorageControlClient;
+  private nextDataClientIndex: number;
+
+  protected constructor(
+    dataClients: IStorageDataClient[],
+    controlClient: IStorageControlClient
+  ) {
+    this.dataClients = dataClients;
+    this.controlClient = controlClient;
+
+    // We round-robin the requests through all of our clients.  Since javascript
+    // is single-threaded, we don't have to worry about thread safety on this
+    // index variable.
+    this.nextDataClientIndex = 0;
+  }
+
+  createStore(storeName: string): Promise<CreateStore.Response> {
+    return this.controlClient.createStore(storeName);
```

Comment:
missing `return await` in these methods?

Reply:
We would like to return the promises and users/client would await for the function like:
```
const listResponse = await storageClient.listStores();
```
(similar for other methods)